src/System.Xml.XmlDocument/src/System/Xml/Dom/XmlElement.cs

```diff
@@ -58,8 +58,14 @@ public override XmlNode CloneNode(bool deep)
                 foreach (XmlAttribute attr in Attributes)
                 {
                     XmlAttribute newAttr = (XmlAttribute)(attr.CloneNode(true));
-                    if (attr is XmlUnspecifiedAttribute && attr.Specified == false)
-                        ((XmlUnspecifiedAttribute)newAttr).SetSpecified(false);
+                    if (attr.Specified == false)
```

Comment:
It's probably fine, but are we sure this change is 100% the same behavior (and at least as good if not better perf) as before?  This changes the order in which things are evaluated and whether they're evaluated at all, e.g. before if attr was not an XmlUnspecifiedAttribute, attr.Specified would never be invoked, but now attr.Specified is invoked first, and only if it's false is the cast done.  I don't know the implementation of these types, so I don't know enough about them to say whether this is identical functionality.